### PR TITLE
prep groups handler for addition of conversations

### PR DESCRIPTION
**Pull request by ryanfkeepers**
Makes modifications to the groups handlers to facilitate the incoming introduction of conversations backup handling. Changes include:
* new handler methods (canMakeDeltas and LocationRef)
* binding handlers to generics
* some naming normalization

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #4536

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Comment by meain** [comment]
You might wanna look into the CI reported issues.

**Comment by ryanfkeepers** [comment]
@meain do you mean the linter issues?  Those will get resolved by stacking the next PR into this one.

**Review by meain — APPROVED** [review]
LGTM. I think we previously had function which were not part of the PR being use which was causing issues in the CI.

File: src/internal/m365/collection/groups/handlers.go [hunk 1]
```diff
@@ -43,11 +51,19 @@ type backupHandler interface {
 		folders *path.Builder,
 		tenantID string,
 	) (path.Path, error)
+
+	locationPath(c C) *path.Builder
+
+	// canMakeDeltaQueries evaluates whether the container can support a
+	// delta query when enumerating its items.
+	canMakeDeltaQueries(C) bool
```
Comment (pandeyabs) [inline src/internal/m365/collection/groups/handlers.go:59]: nit: for consistency

```suggestion
	canMakeDeltaQueries(c C) bool
```

**Review by pandeyabs — APPROVED** [review]

**Pull request merged**